consensus/src/backup/loader.rs

```diff
@@ -63,14 +64,29 @@ impl From<CodecError> for LoaderError {
     }
 }
 
-pub struct BackupLoader<H: Hasher, D: Data, S: Signature, R: Read> {
+#[async_trait]
+pub trait BackupReader {
+    /// Read the entire backup.
+    async fn read(&mut self) -> std::io::Result<Vec<u8>>;
+}
+
+#[async_trait]
+impl<R: Read + Send> BackupReader for R {
```

Comment:
Don't worry about backwards compatibility here, it's better to use actually async implementations if we are changing this interface. If you still need an additional trait (see the comment above) then perhaps replace this implementation with one for `AsyncRead`, otherwise just remove it.

Speaking of compatibility – please bump the version in `Cargo.toml` (to `34.0`?) since this changes the interface.

Reply:
Ah, I forgot to re-read the discussion before commenting, and it turns out to be very relevant. :P Anyway, if you can just use `AsyncRead` as suggested above then it solves the dependency problem, otherwise add it to `types`, as @joschisan suggested.